docs/example/manifests/access/traffic-access-v1.yaml

```diff
@@ -34,12 +34,12 @@ spec:
     methods:
     - GET
     headers:
-    - host: "bookstore.bookstore"
+    - host: "bookstore.bookstore:14001"
     - "user-agent": ".*-http-client/*.*"
     - "client-app": "bookbuyer"
   - name: buy-a-book
     pathRegex: ".*a-book.*new"
     methods:
     - GET
     headers:
-    - host: "bookstore.bookstore"
+    - host: "bookstore.bookstore:14001"
```

Comment:
Rather than specifying this, I suggest removing it entirely because host header matching is already enforced using `virtual_host.domains[]`. 
@snehachhabria added plumbing for this, but specifying host headers here is just going to mislead users to think they can specify whatever they want, when in fact this has to explicitly be a service FQDN:port combination that is already programmed in `virtual_hosts.domains[]`.

More details in https://github.com/openservicemesh/osm/issues/2369#issuecomment-831617932.

Specifying host headers here serves no real purpose in our existing implementation.

Reply:
I agree with @shashankram, there is no real purpose of this in our current implementation given that it has to be the service FQDN:port, which is provided by default in the domains.

Reply:
Makes sense. I see the bookwarehouse HTTPRouteGroup also has a host header match (which doesn't appear to be causing any issues. Should I remove that too?

Reply:
Yes we should remove that as well.

Reply:
> I see the bookwarehouse HTTPRouteGroup also has a host header match (which doesn't appear to be causing any issues. 

I had discussed the removal of this with @snehachhabria a month or so ago, and the only reason I suspect this works is because maestro doesn't fail if requests from the bookstore -> bookwarehouse fail.

Regardless, we should fix this.